aten/src/ATen/native/native_functions.yaml

```
@@ -15729,6 +15729,7 @@
 - func: special_spherical_bessel_j0.out(Tensor x, *, Tensor(a!) out) -> Tensor(a!)
   dispatch:
     CPU, CUDA: special_spherical_bessel_j0_out
+    MPS: spherical_bessel_j0_kernel_mps
```

You are not registering a new function, but rather a new dispatch
```suggestion
    CPU, CUDA, MPS: special_spherical_bessel_j0_out
```